### PR TITLE
Week 7 - Happy Thought - Mai

### PR DESCRIPTION
**Pull request by maikanetaka**
## Netlify link
https://project-happy-thoughts-maikanetaka.netlify.app/

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -1,3 +1,52 @@
...
+  const [thoughts, setThoughts] = useState([]);
+  const [likedPostsCount, setLikedPostsCount] = useState(() => {
+    const savedCount = localStorage.getItem("likedPostsCount");
+    return savedCount ? parseInt(savedCount, 10) : 0;
+  });
+
+  const incrementLikedPostsCount = () => {
+    setLikedPostsCount((prevCount) => {
+      const newCount = prevCount + 1;
+      localStorage.setItem("likedPostsCount", newCount.toString());
```
Comment (almaherris) [inline src/App.jsx:16]: Nice and clean App.jsx! I liked that you used localStorage to store number of likes!

File: src/components/ThoughtForm.jsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    if (error) setError("");
+  };
+
+  const getWordCountText = () => {
+    const typedLength = message.length;
+    const remainingLength = maxLength - typedLength;
+
+    let wordCountMessage;
+    if (remainingLength >= 0) {
+      wordCountMessage = `You have typed ${typedLength} letters, you still can type ${remainingLength} letters.`;
```
Comment (almaherris) [inline src/components/ThoughtForm.jsx:22]: I like this friendly way of helping the user to keep track of the character count, not just 0/140 ☺️

File: src/components/ThoughtForm.jsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    }
+    if (message.length < minLength) {
+      setError(`Message must be at least ${minLength} characters long.`);
+      return;
+    }
+    if (message.length > maxLength) {
+      setError(`Message must be no more than ${maxLength} characters.`);
+      return;
+    }
+
```
Comment (almaherris) [inline src/components/ThoughtForm.jsx:44]: Good way of handling different scenarios!

File: src/components/ThoughtItem.jsx [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    const minutes = Math.round(secondsPast / 60);
+    return minutes === 1 ? "1 minute ago" : `${minutes} minutes ago`;
+  } else if (secondsPast < 86400) {
+    const hours = Math.round(secondsPast / 3600);
+    return hours === 1 ? "1 hour ago" : `${hours} hours ago`;
+  } else {
+    const days = Math.round(secondsPast / 86400);
+    return days === 1 ? "1 day ago" : `${days} days ago`;
+  }
+};
```
Comment (almaherris) [inline src/components/ThoughtItem.jsx:70]: Really good job setting up the time stamp yourself! 🤩  Just FYI (maybe you already know) there are also libraries you can use which will do this for you!

**Review by almaherris — COMMENTED** [review]
Hi Mai! Great job this week, especially after being sick for several days ⭐️ 

I found your code clean and easy to follow. I like that you have split up the app in a lot of components, makes it even easier to understand which part is responsible for what. 

I would just review the responsiveness, since right now it looks very stretched on desktop viewport  🖥

**Comment by dzc1** [comment]
Test

**Comment by dzc1** [comment]
Test

**Review by dzc1 — APPROVED** [review]
Great Job Mai! 

loving to see the topLevel app.jsx comp logic and api calls that are then passed down through props! Great approach as far as implementing the requirements of the project. I will leave you 1 piece of feedback which is related to your function called fetchThoughts on your app.jsx file on line 21. The function structure is great with 1 extra layer that you can add that will enhance it's power. Which is using the async/await syntax. This way you effectively run this call which is calling an 'outsde' resource . 

`const fetchThoughts =  async () => {
   await  fetch("https://happy-thoughts-ux7hkzgmwa-uc.a.run.app/thoughts")
      .then((response) => response.json())
      .then((data) => setThoughts(data))
      .catch((error) => console.error(error));
  };`
. 

This way slight change brings a lot of power.

Great job!

**Pull request closed without merging**